dan_layer/engine/src/runtime/impl.rs

```diff
@@ -660,6 +660,17 @@ impl<TTemplateProvider: TemplateProvider<Template = LoadedTemplate>> RuntimeInte
                         true
                     })?;
 
+                    // add event to indicate that there is a change in component
+                    let (template_address, module_name) =
+                        state.current_template().map(|(addr, name)| (*addr, name.to_string()))?;
+                    state.push_event(Event::new(
```

Comment:
Event is also needed  also happen in SetAccessRules

You could emit the event in `modify_component_with`

```rust
pub fn modify_component_with<F: FnOnce(&mut ComponentHeader) -> bool>(
 //(...snip)
        self.validate_component_state(Some(&before), &after)?;
// Emit an event for the component update
        let (address, _) = self.current_template()?;
        self.push_event(Event::new(
            Some(locked.address().clone()),
            *address,
            self.transaction_hash(),
            std_event("component.updated"),
            Metadata::new(),
        ));

        Ok(())
    }
```

Reply:
Just pushed, thanks!